ls1intum/Artemis
# [Feature] Template and Solution Submission history

### Checklist

### Motivation and Context


We want to have an overview of all submissions of the template and solution participation, to see all results and their corresponding commit hashes.

### Description

The participation-submission data table was adapted to also handle template and solution participations. 
In order to do that, we distinguish between exercises from type programming or other. If it is from type programming, we check if the requested participation is either a template, solution or a regular programming participation.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create a programming exercise
4. Go to the Programming Exercise Details view of the created prog. exercise
5. Scroll down to Template and Solution Result and click on `Show Template Submissions` or `Show Solution Submissions`
6. Verify that the table is shown correctly and also shows the commit hash (which links to the VC's commit page)
7. Next click on the top of the Programming Exercise Details on `Participations` and then on the Submissions counter (navigates to submission table)
8. Verify again that the table is shown correctly and the link of the commit hash works as expected

### Test Coverage

participation-submission-components.ts: 91%

### Screenshots


Programming Exercise Details View with buttons to the submissions page:
![image](https://user-images.githubusercontent.com/33484318/107922741-c231dc80-6f70-11eb-8eb3-299eb735f569.png)


Submissions of Solution Repository (Initial, Changing testcase, updating grading, push changes to solution repo):
![image](https://user-images.githubusercontent.com/33484318/107922685-aa5a5880-6f70-11eb-99f3-3a232062bd8f.png)
